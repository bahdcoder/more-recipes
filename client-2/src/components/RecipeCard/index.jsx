import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import recipePropTypes from '../../propTypes/recipe';

const RecipeCard = ({ recipe }) => (
  <div className="wow fadeIn card mb-3">
    <div className="img-zoom">
      <img
        className="card-img-top"
        style={{ height: 170 }}
        alt=""
        src={recipe.imageUrl}
      />
    </div>
    <div className="card-body">
      <h5 className="card-title h6 text-center">
        <Link to="/recipe/e304eba1-9f87-4424-8c8c-af603976fd23">{recipe.title}</Link>
      </h5>
      <hr />
      <p className="text-sm mb-3">
        <small>
          <span className="text-muted">by </span>
          <Link to="/user/7b3da821-8a11-4238-919e-871c99b1e98f" style={{ textDecoration: 'none' }} className="header-color">{recipe.User.name}</Link>
        </small>
        <span className="text-muted float-right">
          <i className="ion ion-clock mr-2" />
          {recipe.timeToCook} min
        </span>
      </p>
      <p className="text-muted h4 text-center my-2">
        <span className="mr-3 h5">
          <i className="ion ion-recipe-action ion-happy-outline" />
          <span className="ml-3">4k</span>
        </span>
        <span className="mr-3 h5">
          <i className="ion ion-recipe-action ion-sad-outline" />
          <span className="ml-3">2k</span>
        </span>
        <span className="mr-3 h5">
          <i className="ion ion-recipe-action ion-ios-heart-outline" />
          <span className="ml-3">32</span>
        </span>
      </p>
    </div>
  </div>
);

RecipeCard.propTypes = {
  // eslint-disable-next-line react/require-default-props
  recipe: recipePropTypes
};

export default RecipeCard;
