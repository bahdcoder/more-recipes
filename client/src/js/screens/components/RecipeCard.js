import React from 'react';
import { Link } from 'react-router';

import RecipeActions from './RecipeActions';

export default class RecipeCard extends React.Component {
  render() {
    return (
      <div className="card mb-3">
        <div className="img-zoom">
          <img className="card-img-top" style={{height: 250}} src="../../assets/img/meal-1.jpg" alt="Card image cap" />                
        </div>
        <div className="card-body">
          <h5 className="card-title text-center">
            <Link to="/recipe/36515881-5f1b-4f1e-a8eb-0d5fc56171c8">Pressurized African Cassava Golden Grains ( Garri )</Link>
          </h5>
          <hr />
          <p className="text-sm mb-5">
            <small><span className="text-muted">by</span> Elizabeth  Funkirokeze</small>
            <span className="text-muted float-right">
              <i className="ion ion-clock mr-2" />
              35 min</span>
          </p>
          <RecipeActions classes={'h6 text-center'}
                           subClasses={'mr-3 h3'} 
                           {...this.props}/>
        </div>
      </div>
    );
  }
}
