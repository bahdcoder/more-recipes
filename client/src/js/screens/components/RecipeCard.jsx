import React from 'react';
import numeral from 'numeral';
import PropTypes from 'prop-types';
import { Link } from 'react-router';

const RecipeCard = ({ recipe }) => ((
  <div className="wow fadeIn card mb-3">
    <div className="img-zoom">
      <img className="card-img-top" alt="" style={{ height: 170 }} src={recipe.imageUrl} />
    </div>
    <div className="card-body">
      <h5 className="card-title h6 text-center">
        <Link to={`/recipe/${recipe.id}`}>{recipe.title}</Link>
      </h5>
      <hr />
      <p className="text-sm mb-3">
        <small><span className="text-muted">by</span> <Link to={`/user/${recipe.User.id}`} style={{ textDecoration: 'none' }} className="header-color">{recipe.User.name}</Link></small>
        <span className="text-muted float-right">
          <i className="ion ion-clock mr-2" />
          {recipe.timeToCook} min
        </span>
      </p>
      <p className="text-muted h4 text-center my-2">
        <span className="mr-3 h5">
          <i className="ion ion-recipe-action-disabled ion-happy-outline" />
          <span className="ml-3">{numeral(recipe.upvotersIds.length).format('0a')}</span>
        </span>
        <span className="mr-3 h5">
          <i className="ion ion-recipe-action-disabled ion-sad-outline" />
          <span className="ml-3">{numeral(recipe.downvotersIds.length).format('0a')}</span>
        </span>
        <span className="mr-3 h5">
          <i className="ion ion-recipe-action-disabled ion-ios-heart-outline" />
          <span className="ml-3">{numeral(recipe.favoritersIds.length).format('0a')} </span>
        </span>
      </p>
    </div>
  </div>
));

RecipeCard.propTypes = {
  recipe: PropTypes.objectOf(PropTypes.any).isRequired
};

export default RecipeCard;
