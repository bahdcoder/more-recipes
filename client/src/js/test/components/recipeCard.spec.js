/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import renderer from 'react-test-renderer';

import RecipeCard from '../../screens/components/RecipeCard';

import setupTests from '../setupTests';

setupTests();

describe('The RecipeCard component', () => {
  test('should mount without crashing', () => {
    const recipe = {
      id: '3d1b8ff7-b319-4192-b83b-46a0b6357a75',
      title: 'Sit quos quam.',
      userId: '6143df8b-12c3-4c9b-80ca-f9766a74bd4c',
      description: 'Quisquam est veniam iusto.',
      imageUrl: 'http://lorempixel.com/640/480',
      timeToCook: 21763,
      ingredients: '["Laborum ut et dolorem."',
      procedure: '["Molestias ratione omnis nobis tenetur."]',
      User: {
        id: '392887bc-e001-471f-a2ea-d3318e9cead9',
        name: 'Lula Jaskolski',
        email: 'Cristian_Bruen43@gmail.com',
        about: 'atibus ad laborum eos.'
      },
      upvotersIds: [
        'ff2355bf-d16a-4f92-bf4a-2b20c3d7eeca',
        '6aa6d872-3a9e-4022-b4fe-4838dc5e95b7'
      ],
      favoritersIds: ['5d3f1a99-c86a-4c63-a2fc-2257cf7cf4de'],
      downvotersIds: ['7428d9e4-eb13-4a5d-8a93-f9b8c209b61d']
    };
    const wrapper = renderer.create(<RecipeCard recipe={recipe} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
