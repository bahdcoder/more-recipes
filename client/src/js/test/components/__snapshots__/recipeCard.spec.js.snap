// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The RecipeCard component should mount without crashing 1`] = `
<div
  className="wow fadeIn card mb-3"
>
  <div
    className="img-zoom"
  >
    <img
      alt=""
      className="card-img-top"
      src="http://lorempixel.com/640/480"
      style={
        Object {
          "height": 170,
        }
      }
    />
  </div>
  <div
    className="card-body"
  >
    <h5
      className="card-title h6 text-center"
    >
      <a
        onClick={[Function]}
        style={Object {}}
      >
        Sit quos quam.
      </a>
    </h5>
    <hr />
    <p
      className="text-sm mb-3"
    >
      <small>
        <span
          className="text-muted"
        >
          by
        </span>
         
        <a
          className="header-color"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          Lula Jaskolski
        </a>
      </small>
      <span
        className="text-muted float-right"
      >
        <i
          className="ion ion-clock mr-2"
        />
        21763
         min
      </span>
    </p>
    <p
      className="text-muted h4 text-center my-2"
    >
      <span
        className="mr-3 h5"
      >
        <i
          className="ion ion-recipe-action-disabled ion-happy-outline"
        />
        <span
          className="ml-3"
        >
          2
        </span>
      </span>
      <span
        className="mr-3 h5"
      >
        <i
          className="ion ion-recipe-action-disabled ion-sad-outline"
        />
        <span
          className="ml-3"
        >
          1
        </span>
      </span>
      <span
        className="mr-3 h5"
      >
        <i
          className="ion ion-recipe-action-disabled ion-ios-heart-outline"
        />
        <span
          className="ml-3"
        >
          1
           
        </span>
      </span>
    </p>
  </div>
</div>
`;
